[9.0] medor on tour

robinkeunen opened the pull request.

robinkeunen committed:
[ADD] medor_custom: nb_subscribers, nb_cooperators, delivery_point_location

remytms committed:
[ADD] medor_website_ps: Form to edit personal information

robinkeunen committed:
[ADD] mc: trial subscription data, form and controller

houssine78 commented on medor_custom/data/data.xml:
```diff
@@ -37,5 +37,32 @@
             <field name="client_id" ref="data_oauth_provider_client_medor_coop"/>
         </record>
 
+        <record id="data_medor_dummy_trial_journal" model="account.journal">
```
why it doesn't go in demo.xml ?

robinkeunen replied:
Because it is not demo data: it is a fake journal and fake product to make a free trial subscription.

houssine78 replied:
well if it's not for test. putting fake as name of a used product and journal is not a good idea from an accounting point of view. I mean that the administration can be picky on this. 

Why not free_trial  instead of dummy_trial ? It has the advantage to be self explained.

robinkeunen replied:
free_trial is better 👍

houssine78 commented on medor_custom/data/data.xml:
```diff
@@ -37,5 +37,32 @@
+            <field name="refund_sequence" eval="True"/>
+        </record>
+
+        <record id="data_medor_dummy_trial_product" model="product.template">
```
same here

houssine78 commented on medor_custom/models/res_company.py:
```diff
@@ -30,7 +30,7 @@ def _compute_company_data(self):
-                    .search([('member', '=', True)])
+                    .search([('cooperator', '=', True)])
             )
             company.nb_cooperators = len(cooperators)
```
Jean-Claude are you aware that you return the all the res_partner concerned by the easy my coop module and not just the effective cooperator ?

robinkeunen replied:
Mmh, I probably mixed a conflict resolution because I made the mistake before. thx dude.

houssine78 replied:
👍

houssine78 commented on medor_custom/templates/trial_subscription_form.xml:
```diff
@@ -0,0 +1,125 @@
+                                <div class="alert alert-success">
+                                    Your subscription has been successfully registered.
+                                    Check your emails for instructions to set your password.
+                                    <button type="button" class="close" data-dismiss="alert">&amp;times;</button>
```
we can remove this ! it's useless... I know it comes from my shit

robinkeunen replied:
The close button you mean?

houssine78 replied:
yes

houssine78 commented on medor_website_product_subscription/controllers/main.py:
```diff
@@ -0,0 +1,65 @@
+                if key in form.user_fields
+            }
+            user.write(values)
+            if user.parent_id:
```
from my knowledge user doesn't have a parent_id

robinkeunen replied:
@remytms Yep, I think I said in the PR I closed that it should be writing on `user.partner_id`.

houssine78 replied:
👍

remytms replied:
`user` has a `parent_id` field. If I remember well `user` inherit from `partner`.

robinkeunen replied:
^ ce génie

```python
class res_users(osv.osv):
    """ User class. A res.users record models an OpenERP user and is different
        from an employee.

        res.users class now inherits from res.partner. The partner model is
        used to store the data related to the partner: lang, name, address,
        avatar, ... The user model is now dedicated to technical data.
    """
    __uid_cache = {}
    _inherits = {
        'res.partner': 'partner_id',
    }
```

houssine78 reviewed (commented):
need some answers

remytms committed:
[ADD] medor_custom: Form to edit delivery method

This brings a form to add three type of delivery. Delivery to my
personal address, delivery to a friend, deliver to another person.

robinkeunen committed:
[IMP] mc: add depencies to medor custom

houssine78 commented:
i know ;)

Le mer. 4 sept. 2019 à 11:06, Robin Keunen <notifications@github.com> a
écrit :

> *@robinkeunen* commented on this pull request.
> ------------------------------
>
> In medor_custom/data/data.xml
> <https://github.com/coopiteasy/medor/pull/9#discussion_r320651511>:
>
> > @@ -37,5 +37,32 @@
>
>              <field name="client_id" ref="data_oauth_provider_client_medor_coop"/>
>
>          </record>
>
>
>
> +        <record id="data_medor_dummy_trial_journal" model="account.journal">
>
>
> free_trial is better 👍
>
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub
> <https://github.com/coopiteasy/medor/pull/9?email_source=notifications&email_token=AB3IE6WVEA77JGHJ5WATJWDQH53CHA5CNFSM4ITEZT4KYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCDSYBPI#discussion_r320651511>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AB3IE6VJC5LQWI3GF25O2OLQH53CHANCNFSM4ITEZT4A>
> .
>

robinkeunen committed:
fixup! [ADD] mc: trial subscription data, form and controller

robinkeunen committed:
fixup! [ADD] medor_custom: nb_subscribers, nb_cooperators, delivery_point_location

remytms committed:
[ADD] medor_custom: Selection of delivery method in subscription form

remytms committed:
[REF] medor_custom: Selection of delivery method in subscribe form

remytms committed:
[ADD] medor_custom: New routes for trial subcription

remytms committed:
[FIX] medor_custom: Route for delivery method form

remytms committed:
[REF] medor_custom: Adapt delivery method form to new structure

remytms committed:
fixup! [REF] medor_custom: Adapt delivery method form to new structure

remytms committed:
fixup! [ADD] medor_custom: New routes for trial subcription

remytms committed:
[CHG] medor_w_p_subscription: Invoice address save

remytms committed:
[FIX] medor_custom: Invoice address in edit user form

remytms committed:
[FIX] medor_w_p_subscription: Required field in edit user form

robinkeunen committed:
[ADD] product subscription mass mailing dependence

remytms committed:
[FIX] medor_custom: Subscription search in edit user form

remytms committed:
[FIX] medor_custom: No subscription in delivery method form

remytms committed:
[FIX] medor_custom: Delivery method form

This also hide the option to deliver to someone else.

remytms committed:
fixup! [FIX] medor_custom: Delivery method form

remytms committed:
[FIX] medor_custom: Missing sudo in trial form

remytms committed:
[FIX] medor_custom: Trial subscription

remytms committed:
fixup! [FIX] medor_custom: Trial subscription

remytms committed:
[FIX] medor_custom: Missing sudo on delivery method select

robinkeunen committed:
[IMP] medor: redirect from trial sends directly to redirect url
